New Version: Sigil-Ebook.Sigil version 2.1.0

dougmassay opened the pull request:
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/147284)

dougmassay committed:
New Version: Sigil-Ebook.Sigil version 2.1.0

dougmassay commented:
I need just a little more to go on than "issue with installing the application" nothing has changed regarding the installer. It works fine everywhere I've tried it.

stephengillie commented:
Automatic Validation ended with:

> 2024-04-01 18:03:47.191 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\RepositorySource.cpp(757)\WindowsPackageManager.dll!00007FFA4CC1C11F: (caller: 00007FFA4C9A8D6F) LogHr(1) tid(1cf8) 8A15005E     Msg:[D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\Rest\Schema\HttpClientHelper.cpp(30)\WindowsPackageManager.dll!00007FFA4CB22B08: (caller: 00007FFA4CBCBC20) Exception(1) tid(1cf8) 8A15005E ] 
> 2024-04-01 18:03:47.191 [REPO] Failed to open available source: msstore
> 2024-04-01 18:03:54.851 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\RepositorySource.cpp(757)\WindowsPackageManager.dll!00007FFA31D3C11F: (caller: 00007FFA31AC8D6F) LogHr(1) tid(67c) 8A15005E     Msg:[D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\Rest\Schema\HttpClientHelper.cpp(30)\WindowsPackageManager.dll!00007FFA31C42B08: (caller: 00007FFA31CEBC20) Exception(1) tid(67c) 8A15005E ] 
> 2024-04-01 18:03:54.851 [REPO] Failed to open available source: msstore
> 2024-04-01 18:03:56.122 [CLI ] No suitable installer found for manifest Microsoft.VCRedist.2015+.x64 with version 14.38.33135.0

(Automated response - build 876.)

dougmassay commented:
So it can't install the latest Microsoft.VCRedist.2015+.x64 dependency via it's winget manifest? I'm unaware of anything I can do to affect how that particular dependency gets installed. I've named the dependency correctly in my installer yaml. Surely I don't need to specify an exact version of the VCRedist.2015+.x64 package?

dougmassay commented:
Please advise. I'm seeing this error all over the place here. I can't fix a broken VCRedist.2015+.x64 dependency install by pushing more changes to my branch.

dougmassay commented:
#147337

Trenly commented:
@wingetbot run

dougmassay commented:
As mentioned previously, there is nothing for me to "fix." something has changed RE dependency packages that is breaking the Azure Pipeline.

See https://github.com/microsoft/winget-pkgs/issues/147337

No ability to see any detailed logs referenced above - https://github.com/microsoft/winget-pkgs/pull/147284#issuecomment-2030957187 - concerning the error.

stephengillie commented:
Automatic Validation ended with:

> 2024-04-02 05:05:18.989 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\RepositorySource.cpp(757)\WindowsPackageManager.dll!00007FFEE493C11F: (caller: 00007FFEE46C8D6F) LogHr(1) tid(1780) 8A15005E     Msg:[D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\Rest\Schema\HttpClientHelper.cpp(30)\WindowsPackageManager.dll!00007FFEE4842B08: (caller: 00007FFEE48EBC20) Exception(1) tid(1780) 8A15005E ] 
> 2024-04-02 05:05:18.989 [REPO] Failed to open available source: msstore
> 2024-04-02 05:05:26.148 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\RepositorySource.cpp(757)\WindowsPackageManager.dll!00007FFECC74C11F: (caller: 00007FFECC4D8D6F) LogHr(1) tid(18d4) 8A15005E     Msg:[D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\Rest\Schema\HttpClientHelper.cpp(30)\WindowsPackageManager.dll!00007FFECC652B08: (caller: 00007FFECC6FBC20) Exception(1) tid(18d4) 8A15005E ] 
> 2024-04-02 05:05:26.148 [REPO] Failed to open available source: msstore
> 2024-04-02 05:05:27.423 [CLI ] No suitable installer found for manifest Microsoft.VCRedist.2015+.x64 with version 14.38.33135.0

(Automated response - build 876.)

dougmassay commented:
> The issue seems to be a scope mismatch. I believe the dependency only has machine scope, which is conflicting with the user scope here.

There was never a scope mismatch in the past. My yaml hasn't really changed (other than urls, shas, and version numbers) in a long time. If the dependency has changed their scope options, I'm at a loss as to how my package can possibly be installed with the user scope anymore. 

I'd rather not have to assume the necessary vc++ redist libraries are already installed by removing the dependency from my installer yaml altogether.

dougmassay commented:
It could still be a scope issue, but if it is, its not because of any change to the Microsoft.VCRedist.2015+.x64 package itself. It's never had user scope (and no history of it it ever being removed), and my manifest never failed because of it in the past. There has to have been some sort of change in how the InstallerVerification pipeline handles PackageDependencies and multiple scopes.

dougmassay committed:
Test moving dependency to machine scope installer only

dougmassay commented:
It's not at all obvious what sort of feedback is currently being requested of me.

stephengillie commented on manifests/s/Sigil-Ebook/Sigil/2.1.0/Sigil-Ebook.Sigil.installer.yaml:
```diff
@@ -0,0 +1,37 @@
+  InstallerSwitches:
+    Custom: /ALLUSERS /NORESTART /SUPPRESSMSGBOXES /COMPONENTS="dicon,*afiles"
+ManifestType: installer
+ManifestVersion: 1.1.0
```
```suggestion
ManifestVersion: 1.6.0
```

dougmassay replied:
Change made

stephengillie commented on manifests/s/Sigil-Ebook/Sigil/2.1.0/Sigil-Ebook.Sigil.yaml:
```diff
@@ -0,0 +1,8 @@
+PackageVersion: 2.1.0
+DefaultLocale: en-US
+ManifestType: version
+ManifestVersion: 1.1.0
```
```suggestion
ManifestVersion: 1.6.0
```

stephengillie commented on manifests/s/Sigil-Ebook/Sigil/2.1.0/Sigil-Ebook.Sigil.locale.en-US.yaml:
```diff
@@ -0,0 +1,25 @@
+- epub
+ReleaseNotesUrl: https://github.com/Sigil-Ebook/Sigil/releases/tag/2.1.0
+ManifestType: defaultLocale
+ManifestVersion: 1.1.0
```
```suggestion
ManifestVersion: 1.6.0
```

stephengillie reviewed (changes requested):
Sorry for not catching this earlier. We've depreciated this manifest version. But most fields are identical to the most recent version. Increasing the manifest version number is usually enough to pass this check.

dougmassay commented:
It only progressed this far because I moved the Dependencies property from the manifest root to the machine scope installer node to see if a scope mismatch was the culprit. Seems it was. But doing so also made the Manifest-Metadata-Consistency tag appear. I'm not entirely happy with the Dependencies change, but I guess I can live with user scope installs having to manually make sure the dependency is is met before installing. I'll push the requested manifest version changes shortly.

dougmassay committed:
Update manifest schema versions

dougmassay committed:
Move Dependencies property to machine scope installer node

Remove Dependencies property from root of manifest to eliminate a scope mismatch on vcredist dependency. Not an ideal workaround, but I'm tired.
